### PR TITLE
Speed refactor

### PR DESCRIPTION
**Pull request by debugwand**
* make call to get promo data once
* work out from event type whether we want daily, weekly or instant promos
* change fetch promo code to give you a function to fetch the data and a different function to pick out what data you want. For kinesis streams, we don't know which one we want until we are looking at an individual record

### CODE REVIEW

File: functions/create-email/index.js [hunk 1]
```diff
@@ -55,34 +64,56 @@ function processEmail (data) {
 	}));
 }
 
-exports.handle = λ((event) => {
-	logger.info('Got event from ' + getEventSource(event));
+function apiGateway (event, sharedData) {
+	logger.debug(JSON.stringify(event, null, '   '));
```
Comment (laurieboyes) [inline functions/create-email/index.js]: I'm not a fan of using the prettyprinting stuff on `JSON.stringify()` when the logs are going to cloudwatch. It splits each line of the JSON onto its own log line and loses the indentation. Could we change this to just `JSON.stringify(event)`?

File: functions/create-email/test/lib/get-instant-data.spec.js [hunk 2]
```diff
@@ -38,27 +36,17 @@ describe('#getInstantData', () => {
...
-
 	it('gets the email subject', () => {
-		return getInstantData(eventFixture).then(() => {
-			expect(mocks.getInstantSubject.calledOnce).to.be.true;
-		});
+		getInstantData(eventFixture);
+		expect(mocks.getInstantSubject.calledOnce).to.be.true;
 	});
 
 	it('returns a merge of the event, the promo content, and the subject', () => {
```
Comment (laurieboyes) [inline functions/create-email/test/lib/get-instant-data.spec.js]: Should this test case name change to 'returns a merge of the event, ~the promo content~, and the subject'

File: functions/create-email/index.js (hunk 1 above)
Comment (debugwand) [inline functions/create-email/index.js]: this isn't new, this is just moved. Wasn't it there because cloudwatch put everything on one big line?

Comment (laurieboyes) [inline functions/create-email/index.js]: Ha, oh right. Yeah I guess so. I think I prefer it that way but it ain't important. I didn't spot that this was in the ApiGateway bit so will rarely turn up, and even when it does, this isn't too hideous eh. Feel free to disregard.

screen shot 2017-01-13 at 14 37 05

**Pull request merged**